app/services/user_mail_data_processor.rb

```diff
@@ -0,0 +1,32 @@
+class UserMailDataProcessor
+
+  def initialize
+    @encryptor = Encryptor.new(Rails.application.credentials.mail_token_key)
+    @delimiter = ';;;'
+  end
+
+  def get_encrypted_payload(user)
+    sent_at = Time.zone.now
+    decrypted_payload = "#{user.user_name}#{@delimiter}#{sent_at}"
+    @encryptor.encrypt(decrypted_payload)
+  end
+
+  def get_user_params(encrypted_payload)
+    payload = @encryptor.decrypt(encrypted_payload)
+
+    delimiter_index = payload.rindex(@delimiter)
+    user_name = payload[0..delimiter_index-1]
+    created_at = payload[delimiter_index+1..-1]
```

Comment:
Dropping a comment on that because I believe it could be done simpler :)

Reply:
I believe not, as .`split` is not good solution as delimiter can be part of user_name. Any recommendations?